src/m2n/DistributedCommunication.hpp

```diff
@@ -111,12 +111,12 @@ class DistributedCommunication
       int     valueDimension) = 0;
 
   /**
-   * @brief Broadcasts a double to connected ranks on remote participant      
+   * @brief Broadcasts a int to connected ranks on remote participant      
    */
   virtual void broadcastSend(const int &itemToSend) = 0;
 
   /**
-   * @brief Receives a double from a connected rank on remote participant
+   * @brief Receives an int from a connected rank on remote participant
```

Comment:
It's `std::vector<int>`

Reply:
the rank receives one int from the connected ranks and save it into the vector ;)

Reply:
I am not too happy with that design. It's a bit confusing. Why not changing the function to `int` and storing the value in the vector outside of the function.

Reply:
I am not also very happy with that but I think it is the cleanest way:

we need ``_connectionDataVector`` to know we are receiving what from whom. 
In ``M2N.cpp`` class and partition package we do not have access to this data structure. 
of course, we do have access to the connected ranks list in partition package, but if we pass it to the ``broadcasrReceive`` one by one inside a loop, then inside ``PointToPointcom`` we have to use that data (for the the sender rank ). This will look uglier and more difficult to maintain.

Reply:
I see. Let's try to improve the naming and the documentation then. 
Documentation could be "Receives an int per connected rank on remote participant" and then please also document the arguments of the function. 
What about changing the name to `broadcastReceiveAll`?

Reply:
I agree.
I updated the naming and documentation accordingly.